### PR TITLE
Feat/performance nearby

### PR DESCRIPTION
**Pull request by David-Egan**
Docs to be updated by 5pm today to fit changes

### CODE REVIEW

File: openmic/performanceapi/views.py [hunk 1]
```diff
@@ -50,6 +56,134 @@ def create(self, request, *args, **kwargs):
...
+                response_data = {'success': 'False',
+                                 'message': 'User not found. Provide a valid user id.'}
+                return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+
+
+    @api_view(['GET'])
+    def followed_list(self):
+        try:
+            user = Profile.objects.get(pk=self.GET.get('user_id',''))
+        except:
```
Comment (aaron13michael) [inline openmic/performanceapi/views.py]: Should catch exception you are expecting and not all exceptions. If something unexpected happens we need to know.

File: openmic/openmicapi/models.py [hunk 2]
```diff
@@ -7,7 +7,10 @@
 MIN_SEARCH_RADIUS = 10
 
 class Location(models.Model):
-    TEMP_name = models.CharField(max_length=20)
+    latitude = models.DecimalField(max_digits=8, decimal_places=6,
+                                   validators=[MaxValueValidator(90), MinValueValidator(-90)])
+    longitude = models.DecimalField(max_digits=9, decimal_places=6,
+                                    validators=[MaxValueValidator(180), MinValueValidator(-180)])
```
Comment (aaron13michael) [inline openmic/openmicapi/models.py]: We should include fields for name, state, city, address, and zipcode in addition to latitude and longitude so the front-end can receive locations in a human-readable format

File: openmic/openmicapi/migrations/0005_auto_20170817_1543.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+# Generated by Django 1.11.2 on 2017-08-17 15:43
+from __future__ import unicode_literals
+
+import django.core.validators
+from django.db import migrations, models
+
+
+class Migration(migrations.Migration):
```
Comment (aaron13michael) [inline openmic/openmicapi/migrations/0005_auto_20170817_1543.py]: Please don't include migration files in commits. For dev we are just making new ones after each merge into it.

File: openmic/performanceapi/views.py [hunk 4]
```diff
@@ -50,6 +56,134 @@ def create(self, request, *args, **kwargs):
...
             return Response(response_data, status=status.HTTP_201_CREATED, headers=headers)
 
+    @api_view(['GET'])
+    def new_order(self):
+        if self.method == "GET":
+            if Performance.objects.count() == 0:
+                #returns error if there are no performances available to list
+                return Response({"Error":"No performances available."}, status=status.HTTP_400_BAD_REQUEST)
+            else:
+                if 'user_id' in self.GET:
```
Comment (aaron13michael) [inline openmic/performanceapi/views.py]: the front-end should be able to specify a user with their username. I don't think we can expect the client to know a user's primary key in the database. Please update this to either get the primary key given the username or to just work with the username.

File: openmic/performanceapi/views.py [hunk 5]
```diff
@@ -50,6 +56,134 @@ def create(self, request, *args, **kwargs):
...
+                    response_data = {"success":"False",
+                                     "message":"User has no followed artists."}
+                    return Response(response_data, status=status.HTTP_400_BAD_REQUEST)
+                else:
+                    for obj in Performance.objects.filter(profile__in=user.followed_artists.all()):
+                        lst.append(PerformanceSerializer(obj).data)
+                    return Response(lst)
+
+def get_locations_in_radius(request, order_by, current_only, followed_only):
+    user_id = request.GET.get('user_id', '')
```
Comment (aaron13michael) [inline openmic/performanceapi/views.py]: See my above comment about using usernames vs user ids

**Comment by David-Egan** [comment]
Corrections made still working on tests.py as there are somewhat significant changes in models.  I have done some integration testing on my own to verify the changes work with a decent amount of certainty.

**Comment by aaron13michael** [comment]
Looks good. We will move locations to their own module in Sprint 3.

**Pull request merged**